server/src/main/java/org/elasticsearch/script/field/vectors/BFloat16BinaryDenseVectorDocValuesField.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.script.field.vectors;
+
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.index.IndexVersion;
+import org.elasticsearch.index.codec.vectors.BFloat16;
+import org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapper;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.ShortBuffer;
+
+public class BFloat16BinaryDenseVectorDocValuesField extends BinaryDenseVectorDocValuesField {
+    public BFloat16BinaryDenseVectorDocValuesField(
+        BinaryDocValues input,
+        String name,
+        DenseVectorFieldMapper.ElementType elementType,
+        int dims,
+        IndexVersion indexVersion
+    ) {
+        super(input, name, elementType, dims, indexVersion);
+    }
+
+    @Override
+    void decodeDenseVector(IndexVersion indexVersion, BytesRef vectorBR, float[] vector) {
+        ShortBuffer fb = ByteBuffer.wrap(vectorBR.bytes, vectorBR.offset, vectorBR.length).order(ByteOrder.LITTLE_ENDIAN).asShortBuffer();
```

Comment:
Do we need to check for `vectorBR` being null? I see this being done in other fields.

Reply:
I've refactored it a bit and added a check